calculator/interfaces/Main.java

```diff
@@ -0,0 +1,141 @@
+package calculator.interfaces;
+import java.util.Scanner;
+import java.util.InputMismatchException;
+import java.util.regex.*;
+import java.time.ZoneId;
+
+public class Main {
+
+    public static final String basicCalculator = "1";
+    public static final String moduloCalculator = "2";
+    public static final String binaryCalculator = "3";
+    public static final String dateAndTimeCalculator = "4";
+   
+    public static void main(String[] args) {
+        Scanner scan = new Scanner(System.in);
+        String calculatorModel = "0";
+        Pattern pattern;
+        Calculator calculator = null;
+        CalculatorDateAndTime calculatorDateAndTime = null;
+
+        System.out.println("Please chose calculator model");
+        System.out.println("Enter 1 for basic calculator");
+        System.out.println("Enter 2 for modulo calculator");
+        System.out.println("Enter 3 for binary calculator");
+        System.out.println("Enter 4 for Date and Time calculator");
+
+        while(true) {
+            calculatorModel = scan.nextLine();
+
+            if (calculatorModel.equals(basicCalculator)) {
+                calculator = new CalculatorBasic();
+                pattern = Pattern.compile("[\\+\\-\\*\\/]{1}[0-9]+");
+                break;
+            }
+            else if (calculatorModel.equals(moduloCalculator)) {
+                pattern = Pattern.compile("[\\+\\-\\*\\/]{1}[0-9]+");
+                System.out.println("Enter modulo number");
+                calculator = new CalculatorModulo(scan.nextInt());
+                scan.nextLine();
+                break;
+            }
+            else if (calculatorModel.equals(binaryCalculator)) {
+                calculator = new CalculatorBinary();
+                pattern = Pattern.compile("[\\+\\-\\/\\*]{1}[0-1]*");
+                break;
+            }
+            else if (calculatorModel.equals(dateAndTimeCalculator)) {
+                System.out.println("Choose your time zone");
+
+                for (Zone id : Zone.values()) {
+                    System.out.println(id.number + ": " + id);
+                }
+                int city = scan.nextInt();
```

Comment:
that is cityId not city itself, object City containing city information could be named city.